### PR TITLE
Return XMLDocument from document.implementation.createDocument()

### PR DESCRIPTION
**Pull request by domenic**
Currently in the spec document.implementation.createDocument() is specified to return an XMLDocument, not a Document. Additionally, once XMLDocuments can be created, they need to be cloned properly.

Co-Authored-By: ExE Boss <3889017+ExE-Boss@users.noreply.github.com>

Closes #2814 by superseding it.

### CODE REVIEW

File: lib/jsdom/browser/Window.js [hunk 1]
```diff
@@ -119,21 +119,19 @@ function Window(options) {
...
+    encoding: options.encoding,
+    cookieJar: options.cookieJar,
+    url: options.url,
+    lastModified: options.lastModified,
+    referrer: options.referrer,
+    concurrentNodeIterators: options.concurrentNodeIterators,
+    parseOptions: options.parseOptions,
+    defaultView: this._globalProxy,
+    global: this
+  }, { alwaysUseDocumentClass: true });
```
Comment (ExE-Boss) [inline lib/jsdom/browser/Window.js]: Windows with XML documents in them create `XMLDocument` instances.

```suggestion
  });
```

File: lib/jsdom/living/documents.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
+"use strict";
+const XMLDocument = require("../living/generated/XMLDocument.js");
+const Document = require("../living/generated/Document.js");
+const { wrapperForImpl } = require("./generated/utils.js");
+
+exports.createImpl = (globalObject, options, { alwaysUseDocumentClass = false } = {}) => {
+  if (options.parsingMode === "xml" && !alwaysUseDocumentClass) {
```
Comment (ExE-Boss) [inline lib/jsdom/living/documents.js]: If you forget to pass the `options` argument, then you will get a `TypeError`,

File: lib/jsdom/browser/Window.js (hunk 1 above)
Comment (domenic) [reply lib/jsdom/browser/Window.js]: That is incorrect per spec.

File: lib/jsdom/living/documents.js (hunk 2 above)
Comment (domenic) [reply lib/jsdom/living/documents.js]: That seems good.

File: lib/jsdom/browser/Window.js (hunk 1 above)
Comment (ExE-Boss) [reply lib/jsdom/browser/Window.js]: Well, this does result in `true` in all browsers:

```xml
<doc xmlns:html="http://www.w3.org/1999/xhtml">
<html:script><![CDATA[
document.documentElement.append(`${document instanceof XMLDocument}`);
]]></html:script>
</doc>
```

This is necessary for webcompat.

Comment (domenic) [reply lib/jsdom/browser/Window.js]: We won't implement that in jsdom without a spec.

Comment (ExE-Boss) [reply lib/jsdom/browser/Window.js]: Well, <https://github.com/whatwg/html/pull/5104> does fix that.

Comment (domenic) [reply lib/jsdom/browser/Window.js]: If/when that moves forward we can consider it in jsdom.

**Comment by ExE-Boss** [comment]
Well, <https://github.com/jsdom/jsdom/pull/2814> is now equal to this, except it’s been rebased on top of `master`, including the creation of non‑`XMLDocument` instances for XML document windows.

**Comment by ExE-Boss** [comment]
The build fails due to circular dependency issues caused by the execution order placing `XMLDocument‑impl.js` before `Document‑impl.js`, resulting in `DocumentImpl` being `undefined` when `XMLDocument‑impl.js` executes.

I’ve managed to fix this in <https://github.com/jsdom/jsdom/pull/2814>.

**Pull request merged**